src/udev/udevadm-info.c

```diff
@@ -414,6 +420,14 @@ int info_main(int argc, char *argv[], void *userdata) {
                         arg_export = true;
                         arg_export_prefix = optarg;
                         break;
+                case 'w':
+                        arg_wait_for_initialization = true;
+                        if (optarg) {
+                                r = parse_sec(optarg, &arg_wait_for_initialization_timeout);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to parse timeout value: %m");
+                        }
```

Comment:
If no timeout is specified, I think the default value should be infinity, not some magic value.